Project-19-React-router/frontend/src/util/auth.js

```diff
@@ -0,0 +1,40 @@
+import { redirect } from "react-router-dom";
+
+export function getTokenDuration() {
+  const storedExpirationDate = localStorage.getItem("expiration");
+  const expirationDate = new Date(storedExpirationDate);
+  const now = new Date();
+  const duration = expirationDate.getTime() - now.getTime();
+  return duration;
+}
+
+export function getAuthToken() {
+  const token = localStorage.getItem("token");
+
+  if (!token) {
+    return null;
+  }
+
+  const tokenDuration = getTokenDuration();
+
+  if (tokenDuration < 0) {
+    return "EXPIRED";
+  }
+
+  return token;
+}
+
+export function tokenLoader() {
+  const token = getAuthToken();
+  return token;
+}
+
+export function checkAuthLoader() {
+  const token = getAuthToken();
+
+  if (!token) {
+    return redirect("/auth");
+  }
+
+  return token;
+}
```

Comment:
@sknepali The checkAuthLoader function attempts to use redirect from react-router-dom to redirect the user if no token is found. However, redirect is a component and should be used in the render method of a component. Instead, we should throw an error and handle the redirection in the component that calls this function.
```js
try {
  checkAuthLoader();
} catch (error) {
  return <Redirect to="/auth" />;
}
```